dts/bindings/power-domain/intel,adsp-power-domain.yaml

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+description: Power domains for intel ADSP
```

Comment:
I'm seeing DTS bindings being added and a DTS-aware driver being submitted, but... no DTS definitions for a device on the hardware described.  That has my "dead code" sense tingling.  How do we know this works?

Reply:
This was pointed out before, it is basically waiting for mtl board definition gets in to be used. "dead code" is really not the term I would use for this situation, I think it is totally fine submit it to be reviewed and waits the other pr gets merged before merging it.

Reply:
With the board upstream, that should no longer be an issue.